### ISSUE TITLE
Feature Request: LDAP Connector

### ISSUE THREAD

**Issue by tessus** [nextcloud/contacts#454]
I was wondering how complicated it might be to create an LDAP connector, so that one can use the information from the contacts app in clients that allow the use of an LDAP server for looking up addresses.

The Base DN and Bind DN could be hard coded or virtual values. The contacts app could provide a server instance on a separate port to serve LDAP requests (while translating the requests and responses to internal calls and vice versa).

I never really understood why the contacts app didn't use LDAP internally in the first place. This sort of application is one of the main reasons for LDAP anyway.

**Comment by skjnldsv** [comment]
You should post this on server. This app is just a webdav contact editor, such a change should require a direct app from the server :)

**Comment by tessus** [comment]
Well, it's an app for contacts, and since contacts are often stored in LDAP...

I didn't see it as a CardDAV handler only, at least not until now. Why the server? The server can use LDAP for user authentication. While users and entries for addresses can live in the same LDAP, it doesn't mean that the contact organization should be a server thing. An LDAP connector definitely falls into the purview of a contacts app.

**Comment by skjnldsv** [comment]
Because dav is handled by the server. And if we want to create a specific address book generated by ldap, this needs to come from server, or better, the ldap connector app! :)

**Comment by tessus** [comment]
Ah, I see. Since the contacts app gives you the webdav address I originally thought that the app is providing that interface. My bad.

Right, so who's up for writing an LDAP connector app. ;-)

But seriously, wouldn't you say that this would be a nice addition to the nextcloud family?

**Comment by skjnldsv** [comment]
@tessus It could be useful for other people yes :)
There is a discussion about providing a user addressbook from the server, but I don't know if you should post it there or create a new issue on the ldap app?
@blizzz what do you think?

**Comment by blizzz** [comment]
> you should post it there or create a new issue on the ldap app?
@blizzz what do you think?

Rather a new app, LDAP configurations and connections can be reused via `\OCP\LDAPProvider`.

**Comment by tessus** [comment]
So what shall I do? A new app means that I should not create an issue on the ldap app, nor does it mean I should add it to the above mentioned issue.

Where and how would I bring up my idea of an LDAP connector app (access WebDAV contacts via LDAP)?

**Comment by blizzz** [comment]
well, that's a good question, when you don't code. In the end you'll need to find someone to write it. Perhaps you can ask in the [forums](https://help.nextcloud.com/c/feature).

**Comment by tessus** [comment]
Thanks for the info.
